construct/core.py

```diff
@@ -69,6 +69,61 @@ class CancelParsing(ConstructError):
 #===============================================================================
 # used internally
 #===============================================================================
+
+class Context(Container):
+    """Special type of Container used to store contextual information during processing."""
+
+    def __init__(self, _parsing=False, _building=False, _sizing=False, _io=None, _index=0, _subcons=None, **kwargs):
+        super(Context, self).__init__(
+            _=None,                   # Parent context.
+            _root=None,               # Root context.
+            _params=self,             # Global parameters.
+            # TODO: Perhaps have a "state" attribute instead to avoid contradictions due to multiple flags being true.
+            _parsing=_parsing,        # Processing parsing()?
+            _building=_building,      # Processing building()?
+            _sizing=_sizing,          # Processing sizeof()?
+            _io=_io,                  # Processing stream.
+            _index=_index,            # Current index (for Array)
+            _subcons=_subcons or [],  # Current subcons
+        )
+        # Recursively build internal dictionaries as child contexts.
+        for key, value in kwargs.items():
+            if isinstance(value, dict) and value is not self and not isinstance(value, Context):
+                value = self.create_child(**value)
+            self[key] = value
+
+    def create_child(self, _io=None, _subcons=None, **kwargs):
+        """Factory method for initializing a child Context."""
+        # Don't allow children to change the processing status or index
+        kwargs.pop('_parsing', None)
+        kwargs.pop('_building', None)
+        kwargs.pop('_sizing', None)
+        kwargs.pop('_index', None)
+
+        context = Context(
+            _parsing=self._parsing,
+            _building=self._building,
+            _sizing=self._sizing,
+            _io=_io,
+            _subcons=_subcons,
+            _index=self._index,
+            **kwargs,
+        )
+        context._ = self
+        context._params = self._params
+        # First child is root, since the very first parent context holds the user defined external parameters.
+        context._root = self._root or context
+        return context
+
+    def get_child(self, name, default=None):
+        """Retrieves child Context or returns default if it doesn't exist."""
+        child_context = self.get(name, None)
+        if child_context and isinstance(child_context, Context):
+            return child_context
+        else:
+            return default
```

Comment:
This will produce None when the child value is falsey eg. False or [ ] for example.

Reply:
Shouldn't that be expected? A child context should be of type `Context`, and be a truthy value even if the context hasn't been filled. Returning `None` indicates the context has no child.

(sorry for dropping the ball on this, btw)